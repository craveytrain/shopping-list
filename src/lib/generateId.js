import crypto from 'crypto';

const generateId = (name) => crypto.createHash('sha1').update(name).digest('hex');

export default generateId;
