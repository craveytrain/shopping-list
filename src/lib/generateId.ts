import * as crypto from 'crypto';

const generateId = (name: string) => crypto.createHash('sha1').update(name).digest('hex');

export default generateId;
