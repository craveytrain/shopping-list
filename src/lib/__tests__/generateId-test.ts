import generateId from 'lib/generateId';

describe('generateId', () => {
  it('should generate ids', () => {
    const ids = [
      'asdfasdf',
      'aaa adfad f',
      'Hi',
      'aaa adfad f'
    ].map(generateId);

    expect(ids).toMatchSnapshot();
    expect(ids[1]).toEqual(ids[3]);
  });
});
