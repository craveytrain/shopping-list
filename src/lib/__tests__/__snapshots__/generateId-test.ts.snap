// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateId should generate ids 1`] = `
Array [
  "92429d82a41e930486c6de5ebda9602d55c39986",
  "4bf04ba0109e562f9bd0b3df1d59dfca76d1c6a7",
  "94dd9e08c129c785f7f256e82fbe0a30e6d1ae40",
  "4bf04ba0109e562f9bd0b3df1d59dfca76d1c6a7",
]
`;
